### DISCUSSION TITLE
PorousFlowUnsaturated dummy question

### DISCUSSION THREAD

**Discussion by wangshengwa** [idaholab/moose#22194] [Q&A Modules: General] [answered]
Hi all,
I have a simple aquitard-aquifer system with a well in the aquifer that de-saturates the medium. I have made two models one with actions and the other one using kernels. However, I am not getting the same level of saturation at the end of the simulation. Could you please give a hint of what I am doing wrong. Also why the saturation in PorousFlosUnsaturated action is an element variable and with kernels is at the nodes? 

Code: 
https://github.com/wangshengwa/PorousFlow-stuff

Thanks!
Wang

**Comment by WilkAndy** [comment, marked as answer]
Hi @wangshengwa , i'm on holiday so I'm sorry that i can't explore your input files in detail.  I had a quick look and don't see anything obviously different between them.  As for the element-vs-node difference: the saturation is always computed by PorousFlow at the nodes (no matter whether you use an Action or Kernels), however, your `PorousFlowPropertyAux` (note, you correctly used a "constant monomial" saturation variable) will use just one of the nodal values for each element.  Hence, it really only provides an approximate view of the representation held internally by PorousFlow (remember that the finite element solution is only accurate up to the size of the finite element, so this approximation is never too bad).   Perhaps this answers your question (ie, perhaps you were comparing nodal values to elemental values)?

**Comment by wangshengwa** [comment]
I understand now. Thanks a lot, this solves my question.
Wang
